src/main/java/gui/EditRulePopup.java

```diff
@@ -0,0 +1,277 @@
+	private JTextField nameText;
+	private JScrollPane metricsScrollpane;
+	private JComboBox<String> condition;
+
+	private JPanel namePanel;
+	private JPanel metricsPanel;
+	private JPanel metricsListPanel;
+	private JPanel addNewMetricPanel;
+	private JPanel controlPanel;
+
+	/**
+	 * Constructs and initializes the GUI pop-up.
+	 */
+	public EditRulePopup() {
+		frame = new JFrame("Personalized Rules");
+		frame.setLayout(new BorderLayout());
+		frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+		frame.add(createMainPanel());
+		frame.setPreferredSize(new Dimension(685, 300));
+		frame.pack();
+		frame.setLocationRelativeTo(null);
+		frame.setVisible(true);
+	}
+
+	/**
+	 * @return Returns the JPanel where all other JPanels are nested.
+	 */
+	private JPanel createMainPanel() {
+		JPanel mainPanel = new JPanel();
+		mainPanel.setLayout(new BorderLayout());
+		createNamePanel();
+		createMetricsPanel();
+		createControlPanel();
+		mainPanel.add(namePanel, BorderLayout.NORTH);
+		mainPanel.add(metricsPanel, BorderLayout.CENTER);
+		mainPanel.add(controlPanel, BorderLayout.SOUTH);
+		return mainPanel;
+	}
+
+	/**
+	 * Creates and returns the JPanel responsible for holding the rule's name.
+	 * 
+	 * @return
+	 */
+	private JPanel createNamePanel() {
+		namePanel = new JPanel();
+		namePanel.setLayout(new GridLayout(1, 2));
+		JLabel nameLabel = new JLabel("Name: ", SwingConstants.LEFT);
+		nameText = new JTextField();
+		namePanel.add(nameLabel);
+		namePanel.add(nameText);
+		return namePanel;
+	}
+
+	/**
+	 * 
+	 * Creates and returns the JPanel responsible for holding both the current list
+	 * of rule metrics, and the line which allows the addition of more metrics.
+	 * 
+	 * @return
+	 */
+	private JPanel createMetricsPanel() {
+		metricsListPanel = new JPanel();
+		metricsListPanel.setLayout(new BoxLayout(metricsListPanel, BoxLayout.Y_AXIS));
+		metricsScrollpane = new JScrollPane(metricsListPanel, JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+				JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+		fillMetricsListPanel();
+
+		addNewMetricPanel = new JPanel();
+		createAddMetricPanel();
+
+		metricsPanel = new JPanel();
+		metricsPanel.setLayout(new BorderLayout());
+		metricsPanel.add(addNewMetricPanel, BorderLayout.NORTH);
+		metricsPanel.add(metricsScrollpane, BorderLayout.CENTER);
+		return metricsPanel;
+	}
+
+	/**
+	 * Creates and returns the JPanel responsible for holding the JButtons which
+	 * allow all metrics to be cleared, or the rule to be saved.
+	 * 
+	 * @return
+	 */
+	private JPanel createControlPanel() {
+		controlPanel = new JPanel();
+		controlPanel.setLayout(new GridLayout(1, 3));
+		
+		
+		JButton clearButton = new JButton("Clear Metrics");
+		JButton deleteButton = new JButton("Delete Rule");
+		JButton saveButton = new JButton("Save Rule");
+		
+		
+		clearButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				System.out.println("Clear button pressed!");

```

Comment:
does this print appear anywhere? what's the purpose?

Reply:
Print has been deleted.